docs/admin/provisioners.md

```diff
@@ -65,7 +66,68 @@ The provisioner server must authenticate with your Coder deployment. There are t
     --provisioner-tag scope=user
   ```
 
-### Example: Running an external provisioner on a VM
+## Example: Running an external provisioner with Helm
+
+Coder provides a Helm chart for running external provisioner daemons, which you will use in concert with the Helm chart
+for deploying the Coder server.
+
+1. Create a long, random pre-shared key (PSK) and store it in a Kubernetes secret
+
+   ```shell
+   kubectl create secret generic coder-provisioner-psk --from-literal=psk=`head /dev/urandom | tr -dc A-Za-z0-9 | head -c 26`
+   ```
+
+1. Modify your Coder `values.yaml` to include
+
+   ```yaml
+   provisionerDaemon:
+     pskSecretName: "coder-provisioner-psk"
+   ```
+
+1. Redeploy Coder with the new `values.yaml` to roll out the PSK. You can omit `--version <your version>` to also upgrade
+   Coder to the latest version.
+
+   ```shell
+   helm upgrade coder coder-v2/coder \
+       --namespace coder \
+       --version <your version> \
+       --values values.yaml
+   ```
+
+1. Create a `provisioner-values.yaml` file for the provisioner daemons Helm chart. For example
+
+   ```yaml
+   coder:
+     env:
+       - name: CODER_URL
+         value: "https://coder.example.com"
+     replicaCount: 10
+   provisionerDaemon:
+     pskSecretName: "coder-provisioner-psk"
+     tags:
+       location: auh
+       kind: k8s
```

Comment:
It's a bit confusing to me how the provisioner chart and the Coder chart seemingly accept the same Helm values format despite being different charts. If I am understanding correctly, when:

```yaml
provisionerDaemon:
 pskSecretName: "coder-provisioner-psk"
```

is set for the server values, it will set `CODER_PROVISIONER_DAEMON_PSK` for the coder server. I see that code [here](https://github.com/coder/coder/blame/main/helm/coder/templates/_coder.tpl#L38-L44).

However, when the same values file is used set for the provisioner server, it passes the PSK to the provisioner. 

Is this a normal practice for Helm charts? If so, I'm on board. I also wonder, in a follow-up PR, if we should rewrite our workspace proxies Helm to do something similar.

Our workspace proxies Helm is currently on the Coder chart and [when enabled](https://github.com/coder/coder/blob/95d66ac3859fed1b979d7b4891b2af79ab14830a/helm/coder/values.yaml#L135), Helm changes the entrypoint/command to start a wsproxy server instead of a full server. Our current wsproxy approach is easier for me to wrap my head around, but I can see how it may not be a best practice.